### ISSUE TITLE
LITE build undefined reference to units.c++

### ISSUE THREAD

**Issue by Leandros** [capnproto/capnproto#576]
Looks like that the LITE build is now broken. Declaring an object of `MallocMessageBuilder` throws a linker error: 

```
1> libcapnp.lib(layout.obj) : error LNK2019: unresolved external symbol "public: void __thiscall kj::ThrowOverflow::operator()(void)const " (??RThrowOverflow@kj@@QBEXXZ) referenced in function "public: class capnp::Data::Builder __thiscall capnp::_::PointerBuilder::getBlob<struct capnp::Data>(void const *,unsigned int)" (??$getBlob@UData@capnp@@@PointerBuilder@_@capnp@@QAE?AVBuilder@Data@2@PBXI@Z)
```

Edit: It looks like, this is even true for classes derived from `MessageBuilder`. 
Edit 2: Can be fixed by adding `units.c++` to the required build files [here](https://github.com/capnproto/capnproto/blob/master/c%2B%2B/src/kj/CMakeLists.txt#L4)

**Comment by harrishancock** [comment]
@Leandros, sorry for the delay. I presume you're building for Visual Studio? If so, which version?

Commit e2cbc1b5 added code to layout.c++ that calls `assertMax()`, defined in units.h. That file must have had trouble compiling under MSVC at some point, so its implementation file was considered part of the non-lite-mode set of source files for libkj.

@kentonv, now that VS can compile everything, perhaps a good solution to this would be to get rid of the difference in "lite" and "heavy" source code sets? Lite mode could still mean just libcapnp and libkj, but built from their full sources.

**Comment by Leandros** [comment]
I was building with Visual Studio 2017 (Version 15.3.5)

**Comment by harrishancock** [comment]
Thanks @Leandros. I'll resolve this soon, I just have to worry about maintaining compatibility with VS2015, I think. Shouldn't be a big deal.

**Comment by kentonv** [comment]
I tried to take a look at this, but I was unable to build lite mode at all with cmake, as the cmake build seems to ignore the `CAPNP` and `CAPNPC_CXX` environment variables that are supposed to be honored with `EXTERNAL_CAPNP`. I don't understand cmake well enough to figure this out.

Lite mode seems to work with the autotools build, but of course that doesn't support MSVC.

**Comment by kentonv** [comment]
@harrishancock Can you take a look sometime soon so we can fix this for 0.7?

**Comment by harrishancock** [comment]
Yup, will revisit this week.
